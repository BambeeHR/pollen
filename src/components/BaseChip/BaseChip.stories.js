import { boolean, select, text } from '@storybook/addon-knobs';
import BaseChip, { Sizes, Variants } from './BaseChip.vue';
import TypeOverline from '../TypeOverline/TypeOverline.vue';

export default {
  title: 'BaseChip',
  component: BaseChip,
  parameters: {
    componentSubtitle: 'Basic chip component.',
  },
};

const sizes = Object.values(Sizes);
const variants = Object.values(Variants);

export const Gallery = () => ({
  components: { BaseChip, TypeOverline },
  data() {
    return {
      sizes,
      variants,
    };
  },
  template: `
    <div>
      <div class="mb-8" v-for="variant in variants" :key="variant">
        <TypeOverline tag="h2" class="mb-2" variant="large">{{ variant }}</TypeOverline>
        <div class="flex items-end mb-2"> 
          <div v-for="size in sizes" :key="variant + '-' + size" class="mr-8">
            <BaseChip :size="size" :variant="variant">{{ size }}</BaseChip>
          </div>
        </div>
        <div class="flex items-end mb-2"> 
          <div v-for="size in sizes" :key="variant + '-' + size" class="mr-8">
            <BaseChip :size="size" :variant="variant" show-reset>{{ size }}</BaseChip>
          </div>
        </div>
      </div>
    </div>
  `,
});

export const WithKnobs = () => {
  return {
    components: { BaseChip },
    props: {
      text: {
        default: text('Text', 'Hello world'),
      },
      showReset: {
        default: boolean('Show reset', false),
      },
      size: {
        default: select('Size', sizes, Sizes.MEDIUM),
      },
      variant: {
        default: select('Variant', variants, Variants.STANDARD),
      },
    },
    template: `
      <BaseChip
        :show-reset="showReset"
        :size="size"
        :variant="variant"
      >
        {{ text }}
      </BaseChip>
    `,
  };
};

/* eslint-disable no-param-reassign */
[Gallery, WithKnobs].forEach((item) => {
  item.story = {
    parameters: {
      jest: ['BaseChip.test.js'],
    },
  };
});
