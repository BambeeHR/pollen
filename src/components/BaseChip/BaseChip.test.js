import { mount } from '@vue/test-utils';
import BaseChip from './BaseChip.vue';
import { Gallery } from './BaseChip.stories';

describe('BaseChip', () => {
  test('renders correctly', () => {
    const wrapper = mount(BaseChip, {
      slots: {
        default: 'Hello world',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('renders gallery correctly', () => {
    const gallery = Gallery();
    const wrapper = mount(gallery);
    expect(wrapper.element).toMatchSnapshot();
  });
});
