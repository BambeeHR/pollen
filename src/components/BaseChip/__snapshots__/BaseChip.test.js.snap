// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseChip renders correctly 1`] = `
<div
  class="base-chip base-chip--medium base-chip--standard"
  color="inevitable"
>
  <div
    class="base-chip__inner"
  >
    Hello world
  </div>
   
  <!---->
</div>
`;

exports[`BaseChip renders gallery correctly 1`] = `
<div>
  <div
    class="mb-8"
  >
    <h2
      class="type-overline type-overline-large mb-2"
    >
      standard
    </h2>
     
    <div
      class="flex items-end mb-2"
    >
      <div
        class="mr-8"
      >
        <div
          class="base-chip base-chip--small base-chip--standard"
        >
          <div
            class="base-chip__inner"
          >
            small
          </div>
           
          <!---->
        </div>
      </div>
      <div
        class="mr-8"
      >
        <div
          class="base-chip base-chip--medium base-chip--standard"
        >
          <div
            class="base-chip__inner"
          >
            medium
          </div>
           
          <!---->
        </div>
      </div>
      <div
        class="mr-8"
      >
        <div
          class="base-chip base-chip--large base-chip--standard"
        >
          <div
            class="base-chip__inner"
          >
            large
          </div>
           
          <!---->
        </div>
      </div>
    </div>
     
    <div
      class="flex items-end mb-2"
    >
      <div
        class="mr-8"
      >
        <div
          class="base-chip base-chip--small base-chip--standard"
        >
          <div
            class="base-chip__inner"
          >
            small
          </div>
           
          <button
            class="base-chip__reset"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="base-icon svg-inline--fa fa-times fa-w-10 base-chip__reset-icon"
              data-icon="times"
              data-prefix="fal"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class=""
                d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="mr-8"
      >
        <div
          class="base-chip base-chip--medium base-chip--standard"
        >
          <div
            class="base-chip__inner"
          >
            medium
          </div>
           
          <button
            class="base-chip__reset"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="base-icon svg-inline--fa fa-times fa-w-10 base-chip__reset-icon"
              data-icon="times"
              data-prefix="fal"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class=""
                d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="mr-8"
      >
        <div
          class="base-chip base-chip--large base-chip--standard"
        >
          <div
            class="base-chip__inner"
          >
            large
          </div>
           
          <button
            class="base-chip__reset"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="base-icon svg-inline--fa fa-times fa-w-10 base-chip__reset-icon"
              data-icon="times"
              data-prefix="fal"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class=""
                d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="mb-8"
  >
    <h2
      class="type-overline type-overline-large mb-2"
    >
      raised
    </h2>
     
    <div
      class="flex items-end mb-2"
    >
      <div
        class="mr-8"
      >
        <div
          class="base-chip base-chip--small base-chip--raised"
        >
          <div
            class="base-chip__inner"
          >
            small
          </div>
           
          <!---->
        </div>
      </div>
      <div
        class="mr-8"
      >
        <div
          class="base-chip base-chip--medium base-chip--raised"
        >
          <div
            class="base-chip__inner"
          >
            medium
          </div>
           
          <!---->
        </div>
      </div>
      <div
        class="mr-8"
      >
        <div
          class="base-chip base-chip--large base-chip--raised"
        >
          <div
            class="base-chip__inner"
          >
            large
          </div>
           
          <!---->
        </div>
      </div>
    </div>
     
    <div
      class="flex items-end mb-2"
    >
      <div
        class="mr-8"
      >
        <div
          class="base-chip base-chip--small base-chip--raised"
        >
          <div
            class="base-chip__inner"
          >
            small
          </div>
           
          <button
            class="base-chip__reset"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="base-icon svg-inline--fa fa-times fa-w-10 base-chip__reset-icon"
              data-icon="times"
              data-prefix="fal"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class=""
                d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="mr-8"
      >
        <div
          class="base-chip base-chip--medium base-chip--raised"
        >
          <div
            class="base-chip__inner"
          >
            medium
          </div>
           
          <button
            class="base-chip__reset"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="base-icon svg-inline--fa fa-times fa-w-10 base-chip__reset-icon"
              data-icon="times"
              data-prefix="fal"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class=""
                d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="mr-8"
      >
        <div
          class="base-chip base-chip--large base-chip--raised"
        >
          <div
            class="base-chip__inner"
          >
            large
          </div>
           
          <button
            class="base-chip__reset"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="base-icon svg-inline--fa fa-times fa-w-10 base-chip__reset-icon"
              data-icon="times"
              data-prefix="fal"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class=""
                d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="mb-8"
  >
    <h2
      class="type-overline type-overline-large mb-2"
    >
      filled
    </h2>
     
    <div
      class="flex items-end mb-2"
    >
      <div
        class="mr-8"
      >
        <div
          class="base-chip base-chip--small base-chip--filled"
        >
          <div
            class="base-chip__inner"
          >
            small
          </div>
           
          <!---->
        </div>
      </div>
      <div
        class="mr-8"
      >
        <div
          class="base-chip base-chip--medium base-chip--filled"
        >
          <div
            class="base-chip__inner"
          >
            medium
          </div>
           
          <!---->
        </div>
      </div>
      <div
        class="mr-8"
      >
        <div
          class="base-chip base-chip--large base-chip--filled"
        >
          <div
            class="base-chip__inner"
          >
            large
          </div>
           
          <!---->
        </div>
      </div>
    </div>
     
    <div
      class="flex items-end mb-2"
    >
      <div
        class="mr-8"
      >
        <div
          class="base-chip base-chip--small base-chip--filled"
        >
          <div
            class="base-chip__inner"
          >
            small
          </div>
           
          <button
            class="base-chip__reset"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="base-icon svg-inline--fa fa-times fa-w-10 base-chip__reset-icon"
              data-icon="times"
              data-prefix="fal"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class=""
                d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="mr-8"
      >
        <div
          class="base-chip base-chip--medium base-chip--filled"
        >
          <div
            class="base-chip__inner"
          >
            medium
          </div>
           
          <button
            class="base-chip__reset"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="base-icon svg-inline--fa fa-times fa-w-10 base-chip__reset-icon"
              data-icon="times"
              data-prefix="fal"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class=""
                d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="mr-8"
      >
        <div
          class="base-chip base-chip--large base-chip--filled"
        >
          <div
            class="base-chip__inner"
          >
            large
          </div>
           
          <button
            class="base-chip__reset"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="base-icon svg-inline--fa fa-times fa-w-10 base-chip__reset-icon"
              data-icon="times"
              data-prefix="fal"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class=""
                d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
